src/other/precompile.jl

```diff
@@ -0,0 +1,3532 @@
+# precompile.jl contains precompilation directives for all methods whose compilation
+# is triggered by running DataFrames tests and takes more than a given threshold
+# For all=true, the threshold is 0.01, for all=false it is 0.2.
+# `precompile` calls are generated using SnoopCompile via:
+# using SnoopCompileCore
+# inf_timing = @snoopi tmin=0.01 include("test/runtests.jl")
+# using SnoopCompile
+# pc = SnoopCompile.parcel(inf_timing)
+# SnoopCompile.write("src/other/precompile_tmp.jl", pc[:DataFrames], always=true)
+function precompile(all=false)
+    if !all
+        ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
+
+        Base.precompile(Tuple{typeof(DataFrames._combine_tables_with_first!),DataFrame,Tuple{CategoricalArrays.CategoricalArray{String,1,UInt32,String,CategoricalArrays.CategoricalValue{String,UInt32},Union{}}},Array{Int,1},Int,Int,Function,GroupedDataFrame{DataFrame},Nothing,Tuple{Symbol},Val{true}})
```

Comment:
we have many precompilations including CategoricalArrays.jl. I understand they will have to be removed when we drop CategoricalArrays.jl dependency in 1.0. Right?

So maybe better drop them now?

Reply:
Yeah, why not. It's a bit sad that we cannot precompile these without depending on CategoricalArrays, but...